optimize/scalar/brent_min_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright ©2025 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package scalar
+
+import (
+	"math"
+	"testing"
+)
+
+func TestBrentMin(t *testing.T) {
+	tests := []struct {
+		name       string
+		f          func(float64) float64
+		min, max   float64
+		settings   *Settings
+		wantX      float64
+		wantStatus Status
+		distTol    float64 // Tolerance for the distance from wantX
+	}{
+		{
+			name: "Parabola: (x-2)^2 + 1",
+			f: func(x float64) float64 {
+				return (x-2)*(x-2) + 1
+			},
+			min:        0,
+			max:        5,
+			settings:   nil, // Default settings
+			wantX:      2.0,
+			wantStatus: Converged,
+			distTol:    1e-7,
+		},
+		{
+			name: "Cosine: cos(x) in [2, 4]",
+			f: func(x float64) float64 {
+				return math.Cos(x)
+			},
+			min:        2,
+			max:        4,
+			settings:   nil,
+			wantX:      math.Pi, // Minimum is at π
+			wantStatus: Converged,
+			distTol:    1e-7,
+		},
+		{
+			name: "Function: x * cos(x) in [0, 5]",
+			f: func(x float64) float64 {
+				return x * math.Cos(x)
+			},
+			min:        0,
+			max:        5,
+			settings:   nil,
+			wantX:      3.425618459481728, // Approximate solution
+			wantStatus: Converged,
+			distTol:    1e-7,
+		},
+		{
+			name: "Strict Tolerance: (x-2)^2 + 1",
```

Comment:
```suggestion
			name: "strict_tolerance",
```
Names have spaces replaced with underscore, so by doing that in advance, it becomes easier to copy the failing name and find the test in the source.